rpc-interface/src/types.rs

```diff
@@ -178,13 +178,40 @@ impl Block {
         let mut transactions = None;
         if include_body {
             transactions = macro_block.body.as_ref().map(|body| {
-                body.transactions
-                    .iter()
-                    .map(|tx| ExecutedTransaction {
-                        transaction: Transaction::from_reward_tx(tx, block_number),
-                        execution_result: true,
-                    })
-                    .collect()
+                HistoricTransaction::from(
+                    macro_block.network(),
+                    macro_block.block_number(),
+                    macro_block.timestamp(),
+                    vec![],
+                    body.transactions
+                        .iter()
+                        .map(nimiq_transaction::inherent::Inherent::from)
+                        .collect(),
+                    vec![],
+                )
+                .into_iter()
+                .map(|tx| ExecutedTransaction {
+                    transaction: Transaction {
+                        hash: tx.tx_hash().into(),
+                        block_number: Some(macro_block.block_number()),
+                        timestamp: Some(macro_block.timestamp()),
+                        confirmations: cur_block_height.map(|h| h - macro_block.block_number()),
+                        from: Address::default(),
```

Comment:
I missed this in the review, but you changed the sender form the `Policy::COINBASE_ADDRESS` to `Address::default()`, which are different addresses.

Reply:
I fixed it in de902e5ea9f2e6fe6180988977c5081b70d7f35d.

Reply:
I should have alerted you to this, sorry. Is there a reason to prefer the all-zeros address over the coinbase address? As far as I understood it, the `COINBASE` address is the one where new coins come from?

Reply:
I don't understand the question, did you switch the address names in your second sentence?

This code here is converting reward inherents into transactions, so if course the tx should have the coinbase as sender.

Reply:
Actually, I just found this:

https://github.com/nimiq/core-rs-albatross/blob/de902e5ea9f2e6fe6180988977c5081b70d7f35d/primitives/transaction/src/historic_transaction.rs#L300

Can't we use that method here?

Reply:
Woops, you changed it back to `COINBASE` because I accidentally changed it to the all-zeros address. I got confused. Thanks for fixing it.